Automattic/jetpack
# Search: Prepare application for Gutenberg customization

#### Changes proposed in this Pull Request:

Prepares the Jetpack Search Preact application for the upcoming Gutenberg customization interface. Specifically:

* Adds the ability to disable analytics integration.
* Adds the ability to disable DOM integration (mainly related to reading and manipulating the current window's query string).
* Adds support for handling mutable overlay options while preserving support for the WordPress Customizer.
* Registers each search customization option as a [WordPress site setting](https://developer.wordpress.org/reference/functions/register_setting/).
* Refactors initial state generation in PHP into a helper function (`generate_initial_javascript_state`).
* Modularizes `@automattic/color-studio` literalization in Webpack into a separate file. 

#### Jetpack product discussion
pcNPJE-7J-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your Jetpack site with a Search subscription. Alternatively, use Jurassic Ninja to spin up a test site for this branch.
* Perform a search on the site.
  * Ensure that the analytics calls fire as expected; look for `t.gif` requests with a defined `blog_id` query string value.
  * Ensure that the window's query string updates as expected; it should add `?s=` when opening the search modal, for example.
* Customize Jetpack Search options in the Customizer.
  * Ensure that the overlay spawns upon opening the Jetpack Search section of the Customizer.
  * Ensure that each Customization option correctly updates the search overlay. For default sort and excluded post types, the customization iframe should refresh to reflect your changes.

* (bonus) Ensure that `pnpm build` works as expected.
